source/material_model/visco_plastic.cc

```diff
@@ -181,6 +181,8 @@ namespace aspect
                                      const YieldScheme &yield_type,
                                      const std::vector<double> &phase_function_values) const
     {
+      CalculateIsostrainViscosities output_parameters;
+
       // Initialize or fill variables used to calculate viscosities
       std::vector<bool> composition_yielding(volume_fractions.size(), false);
```

Comment:
Instead of initializing new vectors for composition_yielding and composition_viscosities, why not resize the vectors in output_parameters and then fill them directly?